lib/cabin/channel.rb

```diff
@@ -171,9 +188,11 @@ def publish(data, &block)
     event.merge!(@data) # Merge any logger context
 
     @subscriber_lock.synchronize do
-      @subscribers.each do |id, output|
-        append =  block_given? ? block.call(output, event) : true
-        output << event if append
+      @subscribers.each do |_, subscriber|
+        append = block_given? ? block.call(subscriber, event) : true
+        if append && self.class.allow_event?(event, subscriber)
```

Comment:
might be stupid question, but always feel strange about this syntax, is ```self.class``` really necessary here?